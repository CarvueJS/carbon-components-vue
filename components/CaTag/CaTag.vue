<template>
  <span data-tag>
    <slot></slot>
  </span>
</template>

<script>
  const initModifier = ['bx--tag'];

  export default {
    name: 'ca-tag',
    data() {
      return {
        modifier: initModifier,
      };
    },
    props: {
      ibm: {
        type: Boolean,
        default: false,
      },
      beta: {
        type: Boolean,
        default: false,
      },
      thirdParty: {
        type: Boolean,
        default: false,
      },
      local: {
        type: Boolean,
        default: false,
      },
      dedicated: {
        type: Boolean,
        default: false,
      },
      custom: {
        type: Boolean,
        default: false,
      },
      experimental: {
        type: Boolean,
        default: false,
      },
      commumity: {
        type: Boolean,
        default: false,
      },
      private: {
        type: Boolean,
        default: false,
      },
    },
    mounted() {
      if (this.ibm) this.modifier.push('bx--tag--ibm');
      if (this.beta) this.modifier.push('bx--tag--beta');
      if (this.thirdParty) this.modifier.push('bx--tag--third-party');
      if (this.local) this.modifier.push('bx--tag--local');
      if (this.dedicated) this.modifier.push('bx--tag--dedicated');
      if (this.custom) this.modifier.push('bx--tag--custom');
      if (this.experimental) this.modifier.push('bx--tag--experimental');
      if (this.commumity) this.modifier.push('bx--tag--commumity');
      if (this.private) this.modifier.push('bx--tag--private');
      this.$el.className = this.modifier.join(' ');
    },
  };
</script>

<style></style>
