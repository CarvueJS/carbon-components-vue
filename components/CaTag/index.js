import CaTag from './CaTag.vue';

CaTag.install = (Vue) => {
  Vue.component(CaTag.name, CaTag);
};

export default CaTag;
