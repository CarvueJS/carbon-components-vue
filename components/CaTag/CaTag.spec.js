import { shallow } from 'vue-test-utils';
import CaTag from './CaTag.vue';

describe('CaTag', () => {
  test('is a Vue instance', () => {
    const wrapper = shallow(CaTag);
    expect(wrapper.isVueInstance()).toBeTruthy();
  });

  test('Mathes snapshot', () => {
    const wrapper = shallow(CaTag);
    expect(wrapper.html()).toMatchSnapshot()
  });
})
