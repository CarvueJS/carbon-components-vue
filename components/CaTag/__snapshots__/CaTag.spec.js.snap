// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CaTag Mathes snapshot 1`] = `<span data-tag="" class="bx--tag"></span>`;
